frontend/packages/db-structure/src/parser/drizzle/parser.ts

```diff
@@ -0,0 +1,1166 @@
+  sourceCode: string,
+): {
+  tables: Record<string, DrizzleTableDefinition>
+  enums: Record<string, DrizzleEnumDefinition>
+  relations: DrizzleRelationDefinition[]
+} => {
+  try {
+    // Check if TypeScript is available
+    if (!ts || typeof ts.createSourceFile !== 'function') {
+      throw new Error(
+        'TypeScript is not available. Please ensure TypeScript is installed.',
+      )
+    }
+
+    // Use appropriate TypeScript target
+    const scriptTarget = getScriptTarget()
+    const sourceFile = ts.createSourceFile(
+      'schema.ts',
+      sourceCode,
+      scriptTarget,
+      true,
+    )
+
+    const tables: Record<string, DrizzleTableDefinition> = {}
+    const enums: Record<string, DrizzleEnumDefinition> = {}
+    const relations: DrizzleRelationDefinition[] = []
+
+    const visit = (node: ts.Node) => {
+      // Parse pgTable definitions
+      if (ts.isVariableStatement(node)) {
+        const declaration = node.declarationList.declarations[0]
+        if (declaration?.initializer) {
+          // Find the pgTable call in the expression chain
+          const pgTableCall = findPgTableCall(declaration.initializer)
+          if (pgTableCall) {
+            const tableDefinition = parsePgTableCall(declaration, pgTableCall)
+            if (tableDefinition) {
+              tables[tableDefinition.name] = tableDefinition
+            }
+          }
+
+          // Check for other call expressions
+          if (ts.isCallExpression(declaration.initializer)) {
+            const callExpr = declaration.initializer
+
+            // Check if it's a pgEnum call
+            if (
+              ts.isIdentifier(callExpr.expression) &&
+              callExpr.expression.text === 'pgEnum'
+            ) {
+              const enumDefinition = parsePgEnumCall(declaration, callExpr)
+              if (
+                enumDefinition &&
+                declaration.name &&
+                ts.isIdentifier(declaration.name)
+              ) {
+                // Use the variable name as the key for mapping
+                const enumVarName = declaration.name.text
+                enums[enumVarName] = enumDefinition
+              }
+            }
+
+            // Check if it's a relations call
+            if (
+              ts.isIdentifier(callExpr.expression) &&
+              callExpr.expression.text === 'relations'
+            ) {
+              const relationDefinitions = parseRelationsCall(callExpr)
+              relations.push(...relationDefinitions)
+            }
+          }
+        }
+      }
+
+      ts.forEachChild(node, visit)
+    }
+
+    visit(sourceFile)
+    return { tables, enums, relations }
+  } catch (error) {
+    const errorMessage = error instanceof Error ? error.message : String(error)
+    const detailedError = new Error(
+      `Failed to parse Drizzle schema: ${errorMessage}`,
+    )
+    console.error('Error in parseDrizzleSchema:', detailedError)
+    throw detailedError
```

Comment:
If you just want to throw, turn off try-catch and let it throw as it is.

Reply:
I have removed unnecessary try-catch🙏
https://github.com/liam-hq/liam/pull/2299/commits/4d8d776f8d9eb385c82a4c565242bd2f7dc96b11